iothub/device/samples/PnpDeviceSamples/Thermostat/Program.cs

```diff
@@ -2,46 +2,61 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
+using CommandLine;
 using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Provisioning.Client;
+using Microsoft.Azure.Devices.Provisioning.Client.Transport;
 using Microsoft.Azure.Devices.Shared;
 using Microsoft.Extensions.Logging;
-using Newtonsoft.Json;
-using Newtonsoft.Json.Linq;
 
 namespace Thermostat
 {
-    internal enum StatusCode
-    {
-        Completed = 200,
-        InProgress = 202,
-        NotFound = 404,
-        BadRequest = 400
-    }
-
     public class Program
     {
         // DTDL interface used: https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json
         private const string ModelId = "dtmi:com:example:Thermostat;1";
 
-        private static readonly string s_deviceConnectionString = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_CONNECTION_STRING");
-        private static readonly Random s_random = new Random();
-
-        private static DeviceClient s_deviceClient;
         private static ILogger s_logger;
 
-        private static double s_temperature = 0d;
-        private static double s_maxTemp = 0d;
+        public static async Task Main(string[] args)
+        {
+            // Parse application parameters
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams =>
+                {
+                    parameters = parsedParams;
+                })
+                .WithNotParsed(errors =>
+                {
+                    Environment.Exit(1);
+                });
+
+            if (!parameters.Validate())
+            {
+                s_logger.LogError("Required parameters are not set, please recheck required variables by running \"dotnet run -- --help\"");
```

Comment:
If someone has the compiled sample, and not the csproj, `dotnet run` doesn't help them.